Utilities/File.cpp

```diff
@@ -1915,6 +1920,62 @@ fs::file fs::make_gather(std::vector<fs::file> files)
 	return result;
 }
 
+fs::temporary_file::temporary_file()
+{
+	const std::string dir = fs::get_cache_dir() + "/temp-files/";
```

Comment:
Obvious problem: temporary file for this use case should be on the same device, otherwise atomic renaming may be impossible. Can be solved by providing temp path explicitly.
What about cache dir, temp-files, and stuff...
It's more like something that nobody asked for.
`temporary_file` is rather not a very good name for this class, to begin with. It's easily be confused with true temp files which may be found in temp directories but have nothing to do with atomic writing. I'd suggest `pending_file`.

Reply:
Fixed.